File: src/engine/ChangeLens.Core/AnalysisRuns/Models/AnalysisTerminalSummary.cs
namespace ChangeLens.Core.AnalysisRuns.Models;

/// <summary>
///     Represents the immutable strict terminal outcome of a completed analysis run.
/// </summary>
/// <param name="Kind">The strict terminal outcome kind.</param>
/// <param name="TerminalAtUnixMilliseconds">
///     The UTC Unix millisecond timestamp the terminal state committed.
/// </param>
/// <param name="LimitationCount">
///     The non-negative recorded limitation count when <paramref name="Kind" /> is
///     <see cref="AnalysisTerminalKind.CompletedWithLimitations" />; otherwise,
///     <see langword="null" />.
/// </param>
/// <param name="FailureCode">
///     The stable failure code when <paramref name="Kind" /> is
///     <see cref="AnalysisTerminalKind.Failed" />; otherwise, <see langword="null" />.
/// </param>
public sealed record AnalysisTerminalSummary(
    AnalysisTerminalKind Kind,
    long TerminalAtUnixMilliseconds,
    int? LimitationCount,
    string? FailureCode);

File: src/desktop/ui/src/Analysis/Models/AnalysisTerminalSummary.ts
export type AnalysisTerminalSummary =
  | { readonly kind: "completed"; readonly terminalAt: number }
  | {
      readonly kind: "completedWithLimitations";
      readonly terminalAt: number;
      readonly limitationCount: number;
    }
  | { readonly kind: "cancelled"; readonly terminalAt: number }
  | {
      readonly kind: "failed";
      readonly terminalAt: number;
      readonly failureCode: string;
    };
